src/gwt/src/org/rstudio/studio/client/workbench/views/source/SourceColumn.java

```diff
@@ -979,13 +978,14 @@ public void onResponseReceived(SourceDocument newDoc)
                   // apply (dynamic) doc property defaults
                   SourceColumn.applyDocPropertyDefaults(newDoc, true, userPrefs_);
 
-                  EditingTarget target =
-                     addTab(newDoc, Source.OPEN_INTERACTIVE);
+                  EditingTarget target = addTab(newDoc, Source.OPEN_INTERACTIVE);
 
-                  if (contents != null)
+                  // toggle save commands after the editor has been opened,
+                  // in case this action has created and focused a new editor
+                  if (activeEditor_ != null)
                   {
                      target.forceSaveCommandActive();
-                     manageSaveCommands(active);
+                     manageSaveCommands(true);
```

Comment:
The active parameter actually represents whether or not this is this column is the current active column. (`activeEditor_` is only populated for the active column) You'll see in `manageSaveCommands` we use `getNextActiveEditor()` instead of directly accessing the active editor, this is a function that will return the editor that would become active if the column became active. 
When you open a new doc it gets set to the active editor so this code works fine except in the very unlikely case where this function gets called after the user has moved on and is working with a different file. Really it doesn't make a difference, but I think we should pass `this == manager.getActive()` here.

Reply:
Got it -- thanks for explaining!